src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -634,11 +634,13 @@ public void run()
          if (idleTimeout > 0L) {
             logPoolState("Before cleanup\t");
             final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
-            int removable = bag.size() - config.getMinimumIdle();
-            for (PoolBagEntry bagEntry : bag) {
+            final int bagSize = bag.size();
```

Comment:
Because of JIT optimizations, the iterator form of ``for`` is [faster](https://aliteralmind.wordpress.com/2014/03/22/for_foreach/).